### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by Elham-april** [grimme-lab/xtb#684] [General]
Dear friends, I worked and installed ORCA version 5.0.2. Is it necessary to install xtb? can i use CREST without installing xtb?
With Regards

**Comment by awvwgk** [comment]
Please share some information on what you want to do to make it easier for people to answer your question. Regarding Orca, you need an installation of `xtb` for all calculations which use the extended tight binding (xTB) Hamiltonian to work, similar you need an installation of `xtb` to use `crest`.

Depending on your platform, installation instructions differ slightly. Without this information I cannot answer this question other than by referring you to https://xtb-docs.readthedocs.io/en/latest/setup.html for the general instructions.

**Reply by Elham-april** [reply]
Thank you so much. I want to calculate pKa of some flexible molecules, which have carboxylic functional group. I have a problem to find low-energy conformers to find reliable and agreement pKa respect to experimental data. by changing the initial conformer, i had different conformer. Until i found Dr Grimme's paper, attached paper,
[c9cp06869d-new-حتما.pdf](https://github.com/grimme-lab/xtb/files/9482099/c9cp06869d-new-.pdf)
 that introduce CREST package to find low-lying conformer. Could you please help me in installing package? 

Thank you so much
With Regards

**Comment by Elham-april** [comment]
I am waiting for your answer. Could you please answer me? Could you please help me in installing CREST package?

Thank you so much
With Regards

**Reply by awvwgk** [reply]
I have been on vacation until today, please be patient. Also, please consider that many of us are working beside our research or on volunteer basis on this project. I'm happy to help out whenever I can, however this will only work efficiently for both of us if you provide more information on the platform / machine / computer you are using.

**Comment by Elham-april** [comment]
I am waiting for your answer. Could you please answer me? Could you please
help me in installing CREST package? If my project is not related to CREST
package, please please tell me. Please answer to me. I explained my
projection in github.

Thank you so much
With Regards

در تاریخ شنبه ۳ سپتامبر ۲۰۲۲،‏ ۱۱:۳۹ Sebastian Ehlert <
***@***.***> نوشت:

**Comment by awvwgk** [comment]
Since I don't know what platform you are working on, I can only share some general comments with you, which should work for Linux and MacOS.

1. Install [`mambaforge`](https://github.com/conda-forge/miniforge#mambaforge) for Linux/x86\_64 this works by running the following commands in your terminal (the installer is interactive)
```
wget https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh -O Mambaforge.sh
sh Mambaforge.sh
rm Mambaforge.sh
```
2. Restart the terminal
```
exec $SHELL
```
3. Create an environment for `xtb` and `crest` (make sure to pin libblas to MKL for optimal performance on x86\_64 machines)
```
mamba create -n crest crest xtb 'libblas=*=*mkl*'
```
4. Activate the environment and check whether `xtb` and `crest` are available
```
mamba activate crest
xtb --version
crest --version
```
5. Create `otool_xtb` to register `xtb` with Orca (see https://xtb-docs.readthedocs.io/en/latest/setup.html#using-xtb-with-orca)
```
ln -s $(command -v xtb) ${$(command -v orca_scf)/orca_scf/otool_xtb}
```

**Comment by Elham-april** [comment]
Dear Sebastian Ehlert

Thank you so much for your encouraged response and I apologize for my
hurried behaviour.
Linux has not been installed in my system and now I am going to install it.
In your opinion, with below specification of my system, half of the space
is sufficient for installing Linux OR should I install linux with all of my
system space? I mentioned that I am going to install ORCA+xtb+CREST on
Linux. How do I divide my space until in calculation progress, I don't face
inefficient space or memory error?
my specification system is listed below:
Processor(CPU):core i7-6700k cpu @ 4.00 GHZ 4.00 GHZ
Installed memory: 32.0 GB
system type:64bit operating system
I have one driver and its space Drive C: 476 GB

Thank you so much
With Regards

**Reply by awvwgk** [reply]
While you still haven't told me yet, I presume you are on a Windows machine (drive C is a giveaway). This information would have been useful in the initial question. You can use ORCA and xtb on Windows, however CREST does not support this platform. An option is to use Windows Subsystem Linux (WSL) and install the required programs there, you can find the instructions we give to our master students for this at https://qc2-teaching.readthedocs.io/en/latest/install.html#install-wsl, then you can follow the steps shown above.

Also, I cannot give you any advice on how to setup or repartition you computer, that's something you have to learn on your own or ask in a forum which usually deals with this kind of questions. The same holds for any questions regarding the WSL setup, please look into the support provided by your operating system provider.

**Comment by Elham-april** [comment]
Thank you
yes, Windows 10 has been installed on my system.

With Regards

**Comment by Elham-april** [comment]
Dear Sebastian Ehlert

yesterday, I installed linuxmint20.2 on my system, although your link
suggested Ubuntu. Have it any problems during installing and working with
CREST and xtb?

Thank you so much
With Regards

**Reply by awvwgk** [reply]
If you have a specific installation question, there is the possibility to get an answer. For this general question I recommend to just try installing it and see for yourself whether the programs work.

**Comment by Elham-april** [comment]
Exactly, Linuxmint-20.2-cinnamon-64bit

Thank you so much

**Comment by Elham-april** [comment]
Dear friend,

Could you please tell me, does CREST have any restriction on number of
atoms in molecule or xtb in preoptimization? my molecule has 97 C, H and 4
N, and 8 O, named spiro-OMeTAD.

Thank you so much
With Regards

**Reply by awvwgk** [reply]
There are generally no atom restrictions in xtb nor in crest. Please check the setup page of xtb for possible causes. I would assume that you need to add``ulimit -s unlimited`` to your ``.bashrc``/``.zshrc`` to avoid this error.

**Comment by Elham-april** [comment]
Could you please tell me what do i do when i write this commend an give bellow error?
mamba create -n crest crest xtb 'libblas=*=*mkl*'

Command 'mamba' not found, did you mean:

  command 'samba' from deb samba (2:4.13.17~dfsg-0ubuntu1.20.04.1)

Try: sudo apt install <deb name>

**Reply by awvwgk** [reply]
Please install mamba following the instructions in ~~https://github.com/fortran-lang/fpm/pull/806~~ https://github.com/conda-forge/miniforge

**Reply by Elham-april** [reply]
thank you so much
excuse me, i don't understand. could you please explain which check marks in box i should be choosed for accessing instruction?
thank you
regards

**Reply by awvwgk** [reply]
Sorry, copied the wrong link. The mamba installer can be found at https://github.com/conda-forge/miniforge

**Comment by Elham-april** [comment]
Dear Friend,

the address of mamba installer is identical to the below command in the
terminal (actually download mamba):

wget https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh
-O Mambaforge.sh
am i right?

then I do this command:sh Mambaforge.sh

and perform it, in last line of licence agreement has been written
redistribution and use in source and binary forms, with or without. i
used control C.

mamba create -n crest crest xtb 'libblas=*=*mkl*' was performed in
terminal but again was written : command "mamba" not found.

i mention that i add addulimit -s unlimited to my .bashrc/.zshrc and
did again bove command but again was not found mamba.

Could you please help me?

Thank you so much

Regards

**Comment by Elham-april** [comment]
Excuse me, should be used 'enter' instead of 'contl+C.
the answer of "do you wish the installer to initialize mambforg by running
condo inite" is yes or no? i don't know.

Thank you so much
Regards

**Comment by Elham-april** [comment]
I choose yes. after install mambe this error was appeared. Could you please
help me?

An unexpected error has occurred. Conda has prepared the above report.
(base) ***@***.***:~$ mamba activate crest

EnvironmentNameNotFound: Could not find conda environment: crest
You can list all discoverable environments with `conda info --envs`.


(base) ***@***.***:~$     xtb --version

Command 'xtb' not found, did you mean:

  command 'jtb' from deb jtb (1.4.12-2)
  command 'xte' from deb xautomation (1.09-2)
  command 'xtv' from deb xtv (1.1-15)
  command 'xcb' from deb xcb (2.4-5)

Try: sudo apt install <deb name>

(base) ***@***.***:~$     crest --version

Command 'crest' not found, did you mean:

  command 'mrest' from deb libweb-mrest-perl (0.288-1)
  command 'ctest' from deb cmake (3.16.3-1ubuntu1.20.04.1)

Try: sudo apt install <deb name>

**Comment by Elham-april** [comment]
exactly, I take this error: please guide me. thank you so much
(base) ***@***.***:~$ mamba create -n crest crest xtb 'libblas=*=*mkl*'

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ/  /â–ˆâ–ˆ/  /â–ˆâ–ˆ/
 /â–ˆâ–ˆ/
 /â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        â–ˆâ–ˆâ–ˆâ•—   â–ˆâ–ˆâ–ˆâ•— â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ•— â–ˆâ–ˆâ–ˆâ•—
â–ˆâ–ˆâ–ˆâ•—â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ•—  â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ•—
        â–ˆâ–ˆâ–ˆâ–ˆâ•— â–ˆâ–ˆâ–ˆâ–ˆâ•‘â–ˆâ–ˆâ•”â• â•
â–ˆâ–ˆâ•—â–ˆâ–ˆâ–ˆâ–ˆâ•— â–ˆâ–ˆâ–ˆâ–ˆâ•‘â–ˆâ–ˆâ•”â• â• â–ˆâ–ˆâ•—â–ˆâ–ˆâ•”â•
â• â–ˆâ–ˆâ•—

â–ˆâ–ˆâ•”â–ˆâ–ˆâ–ˆâ–ˆâ•”â–ˆâ–ˆâ•‘â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ•‘â–ˆâ–ˆâ•”â–ˆâ–ˆâ–ˆâ–ˆâ•”â–ˆâ–ˆâ•‘â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ•”â•
â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ•‘
        â–ˆâ–ˆâ•‘â•šâ–ˆâ–ˆâ•”â• â–ˆâ–ˆâ•‘â–ˆâ–ˆâ•”â• â•
â–ˆâ–ˆâ•‘â–ˆâ–ˆâ•‘â•šâ–ˆâ–ˆâ•”â• â–ˆâ–ˆâ•‘â–ˆâ–ˆâ•”â• â•
â–ˆâ–ˆâ•—â–ˆâ–ˆâ•”â• â• â–ˆâ–ˆâ•‘
        â–ˆâ–ˆâ•‘ â•šâ• â• â–ˆâ–ˆâ•‘â–ˆâ–ˆâ•‘  â–ˆâ–ˆâ•‘â–ˆâ–ˆâ•‘ â•šâ• â•
â–ˆâ–ˆâ•‘â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ•”â• â–ˆâ–ˆâ•‘  â–ˆâ–ˆâ•‘
        â•šâ• â•     â•šâ• â• â•šâ• â•  â•šâ• â• â•šâ• â•     â•šâ• â•
â•šâ• â• â• â• â• â• â•šâ• â•  â•šâ• â•

        mamba (1.0.0) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ


Looking for: ['crest', 'xtb', 'libblas=[build=*mkl*]']

conda-forge/linux-64                                ??.?MB @  ??.?MB/s 0
failed 10.2s
Download error (28) Timeout was reached [
https://conda.anaconda.org/conda-forge/noarch/repodata.json]
Connection time-out

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/conda/exceptions.py",
line 1129, in __call__
        return func(*args, **kwargs)
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line
935, in exception_converter
        raise e
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line
928, in exception_converter
        exit_code = _wrapped_main(*args, **kwargs)
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line
886, in _wrapped_main
        result = do_call(parsed_args, p)
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line
754, in do_call
        exit_code = create(args, parser)
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line
630, in create
        return install(args, parser, "create")
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line
497, in install
        index = load_channels(pool, channels, repos)
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/utils.py", line
129, in load_channels
        index = get_index(
      File
"/home/elham/mambaforge/lib/python3.10/site-packages/mamba/utils.py", line
110, in get_index
        is_downloaded = dlist.download(api.MAMBA_DOWNLOAD_FAILFAST)
    RuntimeError: Download error (28) Timeout was reached [
https://conda.anaconda.org/conda-forge/noarch/repodata.json]
    Connection time-out

`$ /home/elham/mambaforge/bin/mamba create -n crest crest xtb
libblas=*=*mkl*`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/home/elham/mambaforge/bin/conda
             CONDA_PREFIX=/home/elham/mambaforge
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/home/elham/mambaforge/bin/python
               CONDA_ROOT=/home/elham/mambaforge
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>

 PATH=/home/elham/mambaforge/bin:/home/elham/mambaforge/condabin:/usr/local/

sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/loca
                          l/games:/snap/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>
            XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
         XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0

     active environment : base
    active env location : /home/elham/mambaforge
            shell level : 1
       user config file : /home/elham/.condarc
 populated config files : /home/elham/mambaforge/.condarc
          conda version : 22.9.0
    conda-build version : not installed
         python version : 3.10.6.final.0
       virtual packages : __linux=5.4.0=0
                          __glibc=2.31=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/elham/mambaforge  (writable)
      conda av data dir : /home/elham/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/elham/mambaforge/pkgs
                          /home/elham/.conda/pkgs
       envs directories : /home/elham/mambaforge/envs
                          /home/elham/.conda/envs
               platform : linux-64
             user-agent : conda/22.9.0 requests/2.28.1 CPython/3.10.6
Linux/5.4.0-74-generic linuxmint/20.2 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.

**Comment by awvwgk** [comment]
Sorry, while I'd like to help you, it seems like you are stuck with some more basic installation issues and we are not actually making progress here.

You cannot expect anyone here to make sense of terminal output you copy and paste in this thread. Please, take some time to carefully review and understand the steps you are doing before continuing this discussion.

Also, please don't take offense if me or others prioritize their time on other issues and don't answer your inquiries. We are all just volunteering our time here.

**Comment by Elham-april** [comment]
Dear friend,
Can CREST find the lowest energies of A(+)B(-) complex and dimer molecules?
unfortunately, I can't find these procedures in the CREST site. Could you
please guide me?
for calculating solvation energy in SMD or C-PCM, optimization with
implicit solvent models such as ALPB in XTB and CREST is necessary? because
with and without initial optimization with ALPB have different energies.

Thank you so much
With Regards
